File: Count and Say/Count and Say.cpp
class Solution {
public:
    string countAndSay(int n) {
        string old = "1";
        while (--n > 0) {
            string s = "";
            const char *c = old.c_str();
            for (int i = 0; i < old.size();) {
                int count = 0;
                char cc = c[i];
                while (i < old.size() && cc == c[i]) {
                    ++i;
                    ++count;
                }
                s += to_string(count);
                s += cc;
            }
            old = s;
        }
        return old;
    }
};

File: Count and Say/Count and Say.java
public class Solution {
    public String countAndSay(int n) {
        return countAndSay(n, new HashMap<Integer, String>());
    }
    private String countAndSay(int n, Map<Integer, String> map) {
    	if (n == 1) return "1";
        if (map.containsKey(n)) return map.get(n);
        char[] c = countAndSay(n - 1).toCharArray();
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < c.length;) {
            char cc = c[i];
            int count = 0;
            while (i < c.length && cc == c[i]) {
                ++i;
                ++count;
            }
            sb.append(count);
            sb.append(cc);
        }
        map.put(n, sb.toString());
        return sb.toString();
    }
}